stl/inc/syncstream

```diff
@@ -344,6 +344,7 @@ public:
     basic_osyncstream& operator=(basic_osyncstream&&) noexcept = default;
 
     void emit() {
+        const typename basic_ostream<_Elem, _Traits>::sentry _Ok(*this);

```

Comment:
I think the usual pattern for using a `sentry` is as follows:
```cpp
ios_base::iostate _State = ios_base::goodbit;
const sentry _Ok(*this);

if (!_Ok) {
    _State |= ios_base::badbit;
} else {
    _TRY_IO_BEGIN
    // do work
    _CATCH_IO_END
}

setstate(_State);
return /* whatever */;
```
See [[ostream.unformatted]/1](http://eel.is/c++draft/ostream.unformatted#1) and , e.g., https://github.com/microsoft/STL/blob/d93e386849e812a77258765c38c572166c008c83/stl/inc/ostream#L514-L530

Note that `_CATCH_IO_END` uses `_Myios`, so you'll need to do something like `using _Myios = _Mybase::_Myios;`.

You might be able to test it by explicitly setting `failbit` before calling the function and verifying that `badbit` is set afterwards. Also, if there's data in the `basic_syncbuf` internal buffer, you can verify that it's not written to the wrapped stream.

Also applies to `flush_emit` in `<ostream>`.